src/tests/system/tests/test_netgroups.py

```diff
@@ -309,3 +312,105 @@ def test_netgroup__uid_gt_2147483647(client: Client, provider: GenericProvider):
+        client.sssd.dom("test").update(ad_server=bad_ldap_uri)
+
+    elif isinstance(provider, (IPA)):
+        bad_ldap_uri = "typo.master.%s" % hostname
+        client.sssd.dom("test").update(ipa_server=bad_ldap_uri)
+
+    elif isinstance(provider, (LDAP)):
+        bad_ldap_uri = "ldaps://typo.%s" % hostname
+        client.sssd.dom("test").update(ldap_uri=bad_ldap_uri)
+
+    client.sssd.restart(clean=True)
+
+    # Check backend status
+    assert client.sssd.default_domain is not None, "Failed to load default domain!"
+    assert not client.sssctl.domain_status(client.sssd.default_domain)
+
+    pid_nss = "pidof sssd_nss"
+    pid_nss1 = client.host.conn.run(pid_nss).stdout
+
+    # request for netgroup
+    assert not client.tools.getent.netgroup(netgroup.name)
+
+    pid_nss2 = client.host.conn.run(pid_nss).stdout
+    assert pid_nss1 == pid_nss2, "sssd_nss process id changed!"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.ticket(bz=1779486)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_netgroup__background_refresh(client: Client, provider: GenericProvider):
+    """
+    :title: Verify Netgroup Membership Updates in SSSD Cache After User Addition and Cache Expiry
+    :setup:
+        1. Update SSSD configuration
+        2. Restart SSSD
+        3. Create a user and netgroup
+        4. A second user is created and added to the netgroup
+    :steps:
+        1. The getent command is used to retrieve the netgroup information and user is member of the netgroup
+        2. Wait for 30 seconds to allow the cache to expire and be refreshed
+        3. The ldbsearch command is used to query the SSSD cache database (cache_test.ldb)
+            to verify that second user is now part of the netgroup in the cache
+    :expectedresults:
+        1. Retrieves the netgroup information and user is member of the netgroup
+        2. Successfully wait
+        3. Second user is now part of the netgroup in the cache
+    :customerscenario: True
+    """
+    client.sssd.dom("test").update(entry_cache_timeout="10", refresh_expired_interval="5")
+    client.sssd.restart(clean=True)
+    user = provider.user("user-1").add()
+    netgroup = provider.netgroup("ng-1").add().add_member(user=user)
+
+    result = client.tools.getent.netgroup(netgroup.name)
+    assert result is not None, "Could not get netgroup ng-1"
+    assert result.members[0].user == "user-1"
+
+    user2 = provider.user("user-2").add()
+    netgroup.add_member(user=user2.name)
+
+    time.sleep(30)
+
+    search_result = client.ldb.search("/var/lib/sss/db/cache_test.ldb", "cn=Netgroups,cn=test,cn=sysdb")
+    assert search_result is not None, "Empty search result!"
+    netgrp = search_result.get("name=ng-1,cn=Netgroups,cn=test,cn=sysdb")
+    assert netgrp is not None, "Netgroup ng-1 not present in search result!"
+    triple = netgrp.get("netgroupTriple")
+    assert triple is not None, "Empty tripple!"
+    assert user2.name in triple[1]
```

Comment:
What is this? Should not we check user is in ng-1?

Reply:
Yes that what i am doing here

Reply:
So what is **"netgroupTriple"** ? You are not creating it nor mentioning in the steps.
Can You clarify the steps/results. Why is  there suddenly triple instead of ng-1?